modules/user_accounts/php/edit_user.class.inc

```diff
@@ -199,10 +199,36 @@ class Edit_User extends \NDB_Form
         }
         ////Get the current permissions/////
         $current_permissionids = $user->getPermissionIDs();
-        $permIDs = array();
+        // All permissions are wiped on update.  $permIDs will contain ALL the
+        // user's permissions after an update.  Therefore its structure can be
+        // defined as:
+        // 1) The new permissions granted by the editor, AND
+        // 2) Certain old permissions that the editor is not permitted to alter
+        // In addition, old permissions should be removed if the editor has
+        // removed them and the editor has permissions to remove them.
+        $newPermissions    = array_keys($values['permID']);
```

Comment:
`$values['permID']` might generate a notice it should be put in the if statement line 213 and should be instancited to empty array here